File: Data Structures & Algorithms/longest-strictly-increasing-or-strictly-decreasing-subarray/submission-3.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        incCur = 1
        incMax = 1
        decCur = 1
        decMax = 1
        for i in range(len(nums)-1):
            if nums[i] < nums[i+1]:
                incCur += 1
                incMax = max(incCur, incMax)
                decCur = 1
            elif nums[i] > nums[i+1]:
                decCur += 1
                decMax = max(decCur, decMax)
                incCur = 1
            else:
                decCur = 1
                incCur = 1
        return max(incMax, decMax)

File: Data Structures & Algorithms/remove-element/submission-3.go
func removeElement(nums []int, val int) int {
    k := 0
    for i := 0; i < len(nums); i++ {
        if nums[i] != val {
            nums[k], nums[i] = nums[i], nums[k]
            k += 1
        }
    }
    return k
}
